RTech.CoffeeMachine.Api.Tests/BrewCoffee/Controllers/CoffeeMachineControllerTests.cs

```diff
@@ -24,12 +30,9 @@ public async Task GetBrewStatus_WhenUnrecognizedBrewStatus_ShouldReturn500()
     {
         var brewStatusProvider = new Mock<IBrewStatusProvider>();
         brewStatusProvider.Setup(p => p.GetBrewStatus())
-            .Returns(BrewStatus.None);
+            .ReturnsAsync(BrewStatus.None);
 
-        var response = await GetEndpointResponse(services =>
-        {
-            services.AddSingleton<IBrewStatusProvider>(brewStatusProvider.Object);
-        });
+        var response = await GetEndpointResponse(brewStatusProvider: brewStatusProvider.Object);
```

Comment:
`GetEndpointResponse` was refactored to make mocking both interfaces & the weather `HttpClient` cleaner & easier (see method `GetEndpointResponse` at bottom of this file)